pkg/autoscaler/k8sclient/k8sclient.go

```diff
@@ -209,15 +227,25 @@ func (k *k8sClient) GetClusterStatus() (clusterStatus *ClusterStatus, err error)
 	return clusterStatus, nil
 }
 
-func (k *k8sClient) UpdateReplicas(expReplicas int32) (prevReplicas int32, err error) {
-	prevReplicas, err = k.updateReplicasAppsV1(expReplicas)
+func (k *k8sClient) UpdateReplicas(expReplicas int32) (err error) {
+	for _, target := range k.scaleTargets.targets {
+		_, err := k.UpdateTargetReplicas(expReplicas, target)
+		if err == nil {
```

Comment:
Should check for non-nil instead?
`if err != nil { ...`

Reply:
Indeed !